src/server/hset_family.cc

```diff
@@ -565,14 +565,15 @@ OpResult<vector<string>> OpGetAll(const OpArgs& op_args, string_view key, uint8_
     DCHECK_EQ(pv.Encoding(), kEncodingStrMap2);
     StringMap* sm = GetStringMap(pv, op_args.db_cntx);
 
-    res.resize(sm->Size() * (keyval ? 2 : 1));
+    // Some items could have expired, yet accounted for in Size(), so reserve() might overshoot
+    res.reserve(sm->Size() * (keyval ? 2 : 1));
```

Comment:
this fix does fix the issue #2104 , but check out other places this Size function is used.
f.e I see that hlen returns the Size, and also I see that HRADFIELD uses this function and also ignores the expire time of items so it will return items that passed the ttl.
Dont you think that part of this fix Size should expire the entries?

Reply:
It's problematic to change the semantics of `Size()`, as (a) it's `const`, and (b) developers don't expect such method to be slow.
Indeed I'll search for other similar use cases as you mentioned (good catch!), but I think we should generally prefer `.reserve()` over `.resize()`, unless we have a good reason. It's also faster for most cases as it does not construct the objects only to be reassigned later.

Reply:
I'll follow up on your above comment in separate PRs, there are a few issues that we need to fix here.